core/lib/dal/src/protocol_versions_web3_dal.rs

```diff
@@ -49,4 +52,45 @@ impl ProtocolVersionsWeb3Dal<'_, '_> {
             .await
             .map(|v| v.unwrap())
     }
+
+    pub async fn get_protocol_version_info_by_id(
+        &mut self,
+        version_id: u16,
+    ) -> DalResult<Option<ProtocolVersionInfo>> {
+        let storage_protocol_version = sqlx::query_as!(
+            StorageApiProtocolVersion,
+            r#"
+            SELECT
+                id AS "minor!",
+                timestamp,
+                bootloader_code_hash,
+                default_account_code_hash,
+                evm_emulator_code_hash,
+                upgrade_tx_hash
+            FROM
+                protocol_versions
+            WHERE
+                id = $1
+            "#,
+            i32::from(version_id)
+        )
+        .instrument("get_protocol_version_by_id")
+        .with_arg("version_id", &version_id)
+        .fetch_optional(self.storage)
+        .await?;
+
+        Ok(storage_protocol_version.map(ProtocolVersionInfo::from))
+    }
+
+    pub async fn get_latest_protocol_version_info(&mut self) -> DalResult<ProtocolVersionInfo> {
+        let latest_version = self
+            .storage
+            .protocol_versions_dal()
+            .latest_semantic_version()
+            .await?
+            .unwrap();
```

Comment:
Bikeshedding: I'd rather return `DalResult<Option<_>>` here and turn the `Ok(None)` variant into an error / `unreachable!` on the caller side. This would allow for more "localized" correctness reasoning.